api/provider/aws/service_read.go

```diff
@@ -58,9 +64,9 @@ func (s *ServiceProvider) readService(clusterName, serviceID string) (*ecs.Servi
 		return nil, err
 	}
 
-	for _, service := range output.Services {
-		return service, nil
+	if len(output.Services) == 0 {
+		return nil, fmt.Errorf("ECS service '%s' in cluster '%s' does not exist", serviceID, clusterName)
```

Comment:
It's important we return specific error codes when entities do not exist. This is because the terraform plugin checks for that code specifically. 

```
return nil, errors.Newf(errors.ServiceDoesNotExist, "Service %s does not exist", serviceID)
```

Reply:
✅